playground/css/vite.config.js

```diff
@@ -61,6 +61,21 @@ export default defineConfig({
           },
         ],
       },
+      less: function (id) {
+        if (id.match(/withOpt1.less/)) {
+          return {
+            additionalData: `@primaryColor: red;`,
+          }
+        } else if (id.match(/withOpt2.less/)) {
+          return {
+            additionalData: `@primaryColor: green;`,
+          }
+        } else {
+          return {
+            additionalData: `@primaryColor: gray;`,
+          }
+        }
+      },
```

Comment:
The additionalData takes a `filename` parameter. So this can be written like:
```js
  less: {
    // NOTE: use magic-string for sourcemap support
    additionalData(content, filename) {
      if (filename.match(/withOpt1.less/)) {
        return `@primaryColor: red;` + content
      } else if (id.match(/withOpt2.less/)) {
        return `@primaryColor: green;` + content
      } else {
        return `@primaryColor: gray;` + content
      }
    }
  }
```

Reply:
It's not enough . additionalData only prepends  string to source files, but can not ensure to overwrite all variable value.

Reply:
I don't get what you mean by "ensure to overwrite all variable value". What is different?

Just in case, `additionalData` supports appending and replacing and much more advanced transformations as it's a function.

Reply:
I write a demo: https://github.com/leolord/vite-not-enough/tree/main.

It's built with vite and webpack.  There are two less files, which uses library A and the other uses library B, and the sources are `https://github.com/leolord/vite-not-enough/blob/main/src/pages/pageA/index.less` and `https://github.com/leolord/vite-not-enough/blob/main/src/pages/pageB/index.less`。

I try to set different less variable to the different pages. 

You can see, than vite + additionalData do not modify the primaryColor, but webpack did.

The output are:

```
Vite: https://github.com/leolord/vite-not-enough/blob/main/dist-vite/assets/index-fe677edf.css
Webpack: https://github.com/leolord/vite-not-enough/blob/main/dist-webpack/main.css
```

The purpose of this pr is to set different var for different libraries, and additionalData is not enough.

Reply:
In that case, you can append the variable instead of prepending it. This is the recommended way to override the values in less.
https://lesscss.org/features/#variables-feature-default-variables

Reply:
Yes. You are right. I update the demo .